shell/common/shell.cc

```diff
@@ -320,14 +279,14 @@ Shell::~Shell() {
   PersistentCache::GetCacheForProcess()->RemoveWorkerTaskRunner(
       task_runners_.GetIOTaskRunner());
 
-  vm_->GetServiceProtocol()->RemoveHandler(this);
-
   fml::AutoResetWaitableEvent ui_latch, gpu_latch, platform_latch, io_latch;
 
   fml::TaskRunner::RunNowOrPostTask(
       task_runners_.GetUITaskRunner(),
-      fml::MakeCopyable([engine = std::move(engine_), &ui_latch]() mutable {
-        engine.reset();
+      fml::MakeCopyable([this, &ui_latch]() mutable {
+        ui_latch_.Wait();
```

Comment:
This `Wait` will not get called a lot more frequently by embedders.  I wonder if it would be worth converting it to a spin lock.

Reply:
(This is on the wrong line :))